### PR TITLE
Add configuration to unused files

### PR DESCRIPTION
**Pull request by yenienserrano**
### Description

Add eslint settings to know which export is not being used and to identify unused files.

#### Tasks

- [x] Add linter rule to detect unused exports
- [x] Add knip tool and script to detect unused files
- [x] Fix package scripts for main plugin related to lint the code
- [x] Fix package script form main plugin related to fomat the code with prettier
- [x] Remove unused files
- [x] Move some imports to import from index.{js,ts} instead of specific file
 
### Issues Resolved

- #6748 

### Evidence

<details><summary>Knip</summary>

```console
yarn knip
yarn run v1.22.22
$ knip --files
✂️  Excellent, Knip found no issues.
✨  Done in 4.47s.
```


</details> 


<details><summary>Details</summary>

```console
yarn lint:public | grep 'import/no-unused-modules'
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found      import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found                          import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found         import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1    error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found                                                                                                                                                                                                                                                                                                                                                                                        import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
    1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
     1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1    error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1    error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
  1:1  error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
   1:1   error  Definition for rule 'import/no-unused-modules' was not found  import/no-unused-modules
error Command failed with exit code 1.
```

</details> 


### Test

1. Run the `yarn knip`.
2. In the main plugin you should not have any unused files.
3. Add a file that is not used anywhere and re-run the command
4. The file should appear as unused
5. Smoke test of the application
6. Everything should work as it should

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: plugins/main/package.json [hunk 1]
```diff
@@ -45,12 +45,15 @@
     "test:jest": "node scripts/jest --runInBand",
     "test:jest:runner": "node scripts/runner test",
     "generate:api-data": "node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e \"console.log(require('./package.json').version)\")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration",
-    "prebuild": "node scripts/generate-build-version"
+    "prebuild": "node scripts/generate-build-version",
+    "knip": "npx knip --files"
```
Comment (Desvelao) [inline plugins/main/package.json:49]: **suggestion**: the usage of `npx` in a package.json script should not be neccesary
```suggestion
    "knip": "knip --files"
```

File: plugins/main/package.json [hunk 2]
```diff
@@ -64,9 +67,7 @@
     "react-cookie": "^4.0.3",
     "read-last-lines": "^1.7.2",
     "timsort": "^0.3.0",
-    "typescript": "^5.0.4",
-    "dompurify": "^3.1.3",
-    "jsdom": "16.7.0"
+    "typescript": "^5.0.4"
```
Comment (Desvelao) [inline plugins/main/package.json:70]: **nitpick**: this should be a development dependency.

File: plugins/main/server/routes/wazuh-elastic.ts [hunk 3]
```diff
@@ -38,35 +38,6 @@ export function WazuhElasticRoutes(router: IRouter) {
...
-          tab: schema.string(),
-          pattern: schema.string(),
-        }),
-        body: schema.any(),
-      },
-    },
-    async (context, request, response) =>
-      ctrl.createClusterVis(context, request, response),
-  );
-
```
Comment (Desvelao) [inline plugins/main/server/routes/wazuh-elastic.ts:69]: **nitpick**: if the endpoints are not used anymore and are removed, we should add a reference into the changelog.

File: plugins/wazuh-check-updates/package.json [hunk 4]
```diff
@@ -15,17 +15,34 @@
...
+    "test:jest:runner": "node scripts/runner test",
+    "lint": "eslint \"{public,server,common}/**/*.{js,jsx,ts,tsx,json}\"",
+    "knip": "knip --files"
   },
   "dependencies": {
     "axios": "^1.6.1",
     "md5": "^2.3.0",
-    "node-cron": "^3.0.2"
+    "node-cron": "^3.0.2",
+    "typescript": "^5.5.2"
```
Comment (Desvelao) [inline plugins/wazuh-check-updates/package.json:26]: **nitpick**: this should be a dev dependency.

File: plugins/wazuh-core/package.json [hunk 5]
```diff
@@ -15,18 +15,35 @@
...
+    "knip": "knip --files"
   },
   "dependencies": {
     "axios": "^1.6.1",
     "json2csv": "^4.1.2",
     "jwt-decode": "^3.1.2",
     "md5": "^2.3.0",
-    "node-cron": "^3.0.2"
+    "node-cron": "^3.0.2",
+    "typescript": "^5.5.2"
```
Comment (Desvelao) [inline plugins/wazuh-core/package.json:28]: **nitpick**: this should be a dev dependency.

File: plugins/main/package.json (hunk 2 above)
Comment (yenienserrano) [reply plugins/main/package.json]: Done

File: plugins/wazuh-check-updates/package.json (hunk 4 above)
Comment (yenienserrano) [reply plugins/wazuh-check-updates/package.json]: Done

File: plugins/wazuh-core/package.json (hunk 5 above)
Comment (yenienserrano) [reply plugins/wazuh-core/package.json]: Done

**Comment by JuanGarriuz** [comment]
## Test

### Knip no issues 🟢 
```
yarn knip                             
yarn run v1.22.19
warning Skipping preferred cache folder "/home/node/.cache/yarn" because it is not writable.
warning Selected the next writable cache folder in the list, will be "/tmp/.yarn-cache-1000".
$ knip --files
✂️  Excellent, Knip found no issues.
Done in 8.39s.
```

### Knip issues 🟢 
```
yarn knip
yarn run v1.22.19
warning Skipping preferred cache folder "/home/node/.cache/yarn" because it is not writable.
warning Selected the next writable cache folder in the list, will be "/tmp/.yarn-cache-1000".
$ knip --files
public/react-services/interfaces/no-export.js
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by Desvelao** [comment]
:green_circle: Test

In the main plugin you should not have any unused files:
```console
node@osd:~/kbn/plugins/wazuh$ yarn knip
yarn run v1.22.19
warning Skipping preferred cache folder "/home/node/.cache/yarn" because it is not writable.
warning Selected the next writable cache folder in the list, will be "/tmp/.yarn-cache-1000".
$ knip --files
✂️  Excellent, Knip found no issues.
Done in 11.62s.
```

Adding a unused file:
```console
node@osd:~/kbn/plugins/wazuh$ yarn knip 
yarn run v1.22.19
warning Skipping preferred cache folder "/home/node/.cache/yarn" because it is not writable.
warning Selected the next writable cache folder in the list, will be "/tmp/.yarn-cache-1000".
$ knip --files
common/unused_file.ts
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

**Review by JuanGarriuz — APPROVED** [review]

**Pull request merged**